src/dreyfus_index_updater.erl

```diff
@@ -31,6 +31,7 @@ update(IndexPid, Index) ->
     erlang:put(io_priority, {view_update, DbName, IndexName}),
     {ok, Db} = couch_db:open_int(DbName, []),
     try
+        {ok, ExcludeIdRevs} = purge_index(Db, IndexPid, Index),
```

Comment:
@chguocloudant What is the reason you are collecting this `ExcludeIdRevs`? Would a better approach be just do `clouseau_rpc:delete(IndexPid, Id);` for all docs in `purge_index` function, and then keep `load_docs` function as it was before without any check `case lists:member({Id, Rev}, ExcludeIdRevs) `.

With my suggestion, If a doc is fully purged, it will be deleted in `purge_index` function using `clouseau_rpc:delete`. If a doc is partially purged, we will first also delete it in `purge_index`, and then the doc's new revision should come through changes to the `load_docs` function, and should be update there.